static/scripts/pay.ts

```diff
@@ -185,25 +198,30 @@ export const pay = async (): Promise<void> => {
     table.setAttribute(`data-details-visible`, detailsVisible.toString());
   });
 
-  const { balance, allowance, decimals } = await fetchTreasury();
-  await toggleStatus(balance, allowance, decimals);
+  fetchTreasury().then(({ balance, allowance, decimals }) => {
+    toggleStatus(balance, allowance, decimals);
+  });
```

Comment:
Nice detail to make this async and run in the background instead of blocking the process.